src/main/java/org/openrewrite/staticanalysis/internal/ClassPathUtils.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis.internal;
+
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class ClassPathUtils {
+    public static boolean isAvailable(String className) {
+        try {
+            Class.forName(className);
```

Comment:
Only just now saw [this comment](https://github.com/openrewrite/rewrite-static-analysis/pull/411#issuecomment-2559823283); indeed makes most sense to add those `<provided>` dependencies, which are missing right now after a publish to Maven local.
```xml
  <dependencies>
    <dependency>
      <groupId>org.jetbrains</groupId>
      <artifactId>annotations</artifactId>
      <version>26.0.1</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.openrewrite</groupId>
      <artifactId>rewrite-java</artifactId>
      <version>8.43.0-SNAPSHOT</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.openrewrite.meta</groupId>
      <artifactId>rewrite-analysis</artifactId>
      <version>2.15.0-SNAPSHOT</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.commons</groupId>
      <artifactId>commons-text</artifactId>
      <version>1.13.0</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.openrewrite</groupId>
      <artifactId>rewrite-templating</artifactId>
      <version>1.21.0-SNAPSHOT</version>
      <scope>runtime</scope>
    </dependency>
  </dependencies>
```

Reply:
You should always use this form and set the `initialize` boolean to `false` so that static initializers etc don't get executed unintentionally.

<img width="1051" alt="image" src="https://github.com/user-attachments/assets/9f3b6ea5-23d4-41f8-991b-75760a096d34" />

> is if the classloader used here would contain the C# / Groovy / Kotlin classes we are likely to parent class load in the CLI & Platform

@timtebeek yes it would, because by default `forName` uses the classloader of `ClassPathUtils`, which will be in the same classloader as the recipe whose parent classloader contains the language specific classes.

Reply:
I would much prefer this utility goes in `openrewrite/rewrite`'s `ReflectionUtils` as it is likely to be used in other recipe modules.

<img width="666" alt="image" src="https://github.com/user-attachments/assets/7d91e97d-7432-4871-8204-919fac864b47" />

Reply:
Done:
- https://github.com/openrewrite/rewrite/pull/4810